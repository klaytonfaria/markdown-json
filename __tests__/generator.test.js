jest.mock('../lib/plugins/transform-data')
const transformDataPlugin = require('../lib/plugins/transform-data')
const { build } = require('../lib/generator')

const {
  SETTINGS,
  SETTINGS_WITH_METADATA
} = require('../stubs/settings')
const { formatBytes } = require('../lib/utils')

describe('Generator tests', () => {
  describe('[plugins]', () => {

    afterEach(() => {
      jest.deepUnmock('../lib/plugins/transform-data')
    })

    it('should call transformData plugin', async () => {
      transformDataPlugin.mockImplementation(() => {
        return jest.fn().mockName('transformData').mockImplementation(files => files)
      })

      await build(SETTINGS)

      // Plugin initializer
      expect(transformDataPlugin).toHaveBeenCalledWith(SETTINGS)

      // Plugin instance
      const pluginInstance = transformDataPlugin.mock.results[0].value
      expect(pluginInstance.getMockName()).toBe('transformData')
      expect(pluginInstance.mock.calls.length).toBe(1)
    })

    it('should transform data to match snapshot', async () => {
      const transformDataPlugin = require('../lib/plugins/transform-data')
      const parsedFiles = await build(SETTINGS)
      const result = transformDataPlugin(SETTINGS)(parsedFiles)

      expect(result).toMatchSnapshot()
    })

    it('should transform data with the metadata property', async () => {
      const transformDataPlugin = require('../lib/plugins/transform-data')
      const parsedFiles = await build(SETTINGS)
      const result = transformDataPlugin(SETTINGS_WITH_METADATA)(parsedFiles)

      expect(result).toHaveProperty('[0].meta')
      expect(new Date(result[0].meta.createdAt)).not.toBeNaN()
      expect(new Date(result[0].meta.lastModified)).not.toBeNaN()
      expect(typeof result[0].meta.size).toBe('number')
      expect(result[0].meta.formattedSize).toBe(formatBytes(result[0].meta.size))
    })
  })
})
