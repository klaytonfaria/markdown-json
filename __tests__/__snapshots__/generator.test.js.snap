// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator tests [plugins] should transform data to match snapshot 1`] = `
Array [
  Object {
    "contents": "<p>Follow some application examples of buttons</p>
<h1 id=\\"types\\">Types</h1>
<h3 id=\\"base\\">Base</h3>
<p>Base button layout sample:</p>
<button type=\\"button\\" class=\\"buy-button btn btn-success\\">
  <span class=\\"icon\\"></span>
  <span class=\\"text\\">Button</span>
</button>

<pre><code class=\\"language-scss\\">.btn-primary {
  @include states(#1A75CE, #086B9C);
}</code></pre>
<pre><code class=\\"language-html\\">&lt;button type=&quot;button&quot; class=&quot;buy-button btn btn-success&quot;&gt;
  &lt;span class=&quot;icon&quot;&gt;&lt;/span&gt;
  &lt;span class=&quot;text&quot;&gt;Button&lt;/span&gt;
&lt;/button&gt;</code></pre>
",
    "device": Array [
      "desktop",
      "mobile",
    ],
    "excerpt": "<p>Follow some application examples of buttons</p>",
    "id": "content__buttons",
    "section": "Elements",
    "styles": Array [
      "https://lalao.com/styles/structure.min.css",
      "https://lalao.com/styles/app.min.css",
    ],
    "title": "buttons",
  },
  Object {
    "contents": "<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>
",
    "date": 2012-12-07T00:00:00.000Z,
    "device": Array [
      "desktop",
      "tablet",
    ],
    "excerpt": "<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>",
    "id": "content__globals__js-utils",
    "section": "Globals",
    "title": "utils [js]",
  },
  Object {
    "contents": "<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>
",
    "date": 2012-12-07T00:00:00.000Z,
    "device": Array [
      "desktop",
      "tablet",
    ],
    "excerpt": "<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>",
    "id": "content__globals__sass-reset",
    "section": "Globals",
    "title": "sass reset",
  },
  Object {
    "contents": "<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>
",
    "date": 2012-12-07T00:00:00.000Z,
    "device": Array [
      "desktop",
      "tablet",
    ],
    "excerpt": "<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>",
    "id": "content__globals__sass-utils",
    "section": "Globals",
    "title": "sass utils [scss]",
  },
  Object {
    "contents": "<h1 id=\\"icons\\">Icons</h1>
<p>Our icons list still is empty :(</p>
",
    "excerpt": "<p>Our icons list still is empty :(</p>",
    "id": "content__icons",
    "section": "Elements",
    "tags": Array [
      "icons",
      "base",
    ],
    "title": "icons",
  },
  Object {
    "contents": "<h2 id=\\"lorem-ipsum-dolor-sit-amet-consectetur-adipisicing-elit\\">Lorem ipsum dolor sit amet, consectetur adipisicing elit.</h2>
<p>Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>
",
    "device": Array [
      "desktop",
      "tablet",
    ],
    "excerpt": "<p>Ullam aspernatur et natus saepe fugiat, eligendi veniam temporibus accusamus molestiae expedita unde? Totam necessitatibus odit consequatur fugit voluptatem, accusamus qui, voluptas.</p>",
    "id": "content__input",
    "section": "Elements",
    "title": "input",
  },
  Object {
    "contents": "<h2 id=\\"o-incentivo-ao-avanço-tecnológico\\">O incentivo ao avanço tecnológico.</h2>
<p>O incentivo ao avanço tecnológico, assim como a constante divulgação das informações ainda não demonstrou convincentemente que vai participar na mudança dos níveis de motivação departamental.
<img src=\\"https://source.unsplash.com/random\\" alt=\\"unsplash\\" height=\\"800\\" width=\\"600\\" />
A certificação de metodologias que nos auxiliam a lidar com a expansão dos mercados mundiais causa impacto indireto na reavaliação do remanejamento dos quadros funcionais.
Podemos já vislumbrar o modo pelo qual a complexidade dos estudos efetuados promove a alavancagem dos níveis de motivação departamental.
Acima de tudo, é fundamental ressaltar que o entendimento das metas propostas é uma das consequências dos procedimentos normalmente adotados.
No entanto, não podemos esquecer que o fenômeno da Internet aponta para a melhoria de alternativas às soluções ortodoxas.
Evidentemente, a adoção de políticas descentralizadoras agrega valor ao estabelecimento do orçamento setorial.
Pensando mais a longo prazo, a adoção de políticas descentralizadoras aponta para a melhoria das condições financeiras e administrativas exigidas.
É importante questionar o quanto a constante divulgação das informações promove a alavancagem dos índices pretendidos.
Evidentemente, a mobilidade dos capitais internacionais acarreta um processo de reformulação e modernização do remanejamento dos quadros funcionais.
O que temos que ter sempre em mente é que o novo modelo estrutural aqui preconizado ainda não demonstrou convincentemente que vai participar na mudança dos métodos utilizados na avaliação de resultados.
No mundo atual, a expansão dos mercados mundiais talvez venha a ressaltar a relatividade das regras de conduta normativas.
Todas estas questões, devidamente ponderadas, levantam dúvidas sobre se o acompanhamento das preferências de consumo ainda não demonstrou convincentemente que vai participar na mudança dos índices pretendidos.</p>
",
    "excerpt": "",
    "id": "content__text-and-image",
    "section": "Content",
    "tags": Array [
      "image",
      "text",
      "long",
    ],
    "title": "Text and image",
  },
]
`;
